src/scene_server.py

```diff
@@ -45,9 +52,9 @@ class ProcessData:
     pipe: AioConnection = None
 
 
-def calculate_deltas(from_tick_number: int, to_tick_number: int, actions_in_ticks: List[List[Union[str]]]) -> List[List[str]]:
+def calculate_deltas(from_tick_number: int, to_tick_number: int, actions_in_ticks: List[List[str]]) -> List[str]:
```

Comment:
`List[List[Optional[str]]] -> List[Optional[str]]`
`List[List[Optional[str]]]`, because we start with 0 applied actions
`List[Optional[str]]`, because with `from_tick_number = x` and  `to_tick_number = x` we return `[]`

Reply:
But an empty list is also a list. It's not a list of elements which are optional. Elements are always of type string. I might be wrong :)

Reply:
They are not necessarily of type string